lib/internal/event_target.js

```diff
@@ -597,19 +597,40 @@ class EventTarget {
     if (arguments.length < 2)
       throw new ERR_MISSING_ARGS('type', 'listener');
 
-    // We validateOptions before the validateListener check because the spec
-    // requires us to hit getters.
-    const {
-      once,
-      capture,
-      passive,
-      signal,
-      isNodeStyleListener,
-      weak,
-      resistStopPropagation,
-    } = validateEventListenerOptions(options);
-
-    validateAbortSignal(signal, 'options.signal');
+    let once = false;
+    let capture = false;
+    let passive = false;
+    let isNodeStyleListener = false;
+    let weak = false;
+    let resistStopPropagation = false;
+
+    if (options !== kEmptyObject) {
+      // We validateOptions before the validateListener check because the spec
+      // requires us to hit getters.
+      options = validateEventListenerOptions(options);
+
+      once = options.once;
+      capture = options.capture;
+      passive = options.passive;
+      isNodeStyleListener = options.isNodeStyleListener;
+      weak = options.weak;
+      resistStopPropagation = options.resistStopPropagation;
+
+      const signal = options.signal;
+
+      validateAbortSignal(signal, 'options.signal');
+
+      if (signal) {
+        if (signal.aborted) {
+          return;
+        }
+        // TODO(benjamingr) make this weak somehow? ideally the signal would
```

Comment:
This todo isn't relevant for like a year or two since this already has kWeakHandler: this